plugins/welcome.lua

```diff
@@ -0,0 +1,124 @@
+function EmptyView:get_name() if welcomed then return old_get_name(self) end return "Welcome!" end
+
+local old_draw = EmptyView.draw
+function EmptyView:draw()
+  if loading then
+    local y = self.position.y + self.size.y / 2
+    self:draw_background(style.background)
+    PluginView.draw_loading_screen(self, loading.label, loading.percent)
+    -- common.draw_text(style.big_font, style.dim, "Installing addons package. Please wait...", "center", self.position.x, y, self.size.x, style.font:get_height())
+    return
+  end
+  old_draw(self)
+  if welcomed then return end
+  local x, y, w, h = self.position.x + self.size.x / 2
+
+  local button_width = math.min(self.size.x / 2 - 80, 300)
+  x = self.position.x + self.size.x / 2 - 50
+  local button_height = style.padding.y * 2 + style.font:get_height()
+  local y = self.position.y + self.size.y / 2 - ((button_height + style.padding.y) * #buttons) / 2 - style.padding.y
+  renderer.draw_rect(x, y, button_width, #buttons * (button_height + style.padding.y), style.background)
+  for i,v in ipairs(buttons) do
+    v.x, v.y, v.w, v.h = draw_button(self, x + style.padding.x, y, button_width, v)
+    y = y + v.h + style.padding.y * 2
+  end
+
+  if hovered_button then
+    for i, v in ipairs(hovered_button.tooltip) do
+      common.draw_text(style.font, style.dim, v, "center", self.position.x, y, self.size.x, style.font:get_height())
+      y = y + style.font:get_height()
+    end
+  else
+    common.draw_text(style.font, style.dim, "Hover over one of the options below to get started.", "center", self.position.x, y, self.size.x, style.font:get_height())
+  end
+end
+
+function EmptyView:on_mouse_moved(x, y)
+  hovered_button = nil
+  for i,v in ipairs(buttons) do
+    if v.x and x >= v.x and x < v.x + v.w and y >= v.y and y < v.y + v.h then
+      hovered_button = v
+    end
+  end
+end
+
+function EmptyView:on_mouse_released(button, x, y)
+  if hovered_button and not welcomed then command.perform(hovered_button.command) end
+end
+
+
+
+local function terminate_welcome()
+  io.open(USERDIR .. PATHSEP .. "welcomed", "wb"):close()
+  welcomed = true
+end
+
+command.add(EmptyView, {
+  ["welcome:install-addons"] = function()
+    core.log("Installing addons...")
+    loading = { percent = 0, label = "Initializing..." }
+    core.redraw = true
+    PluginManager:install({ id = "meta_addons" }, { progress = function(progress) 
+      loading = progress 
+      core.redraw = true
+    end, restart = false }):done(function()
+      loading = false
+      core.log("Addons installed!")
+      terminate_welcome()
+      command.perform("core:restart")
+    end):fail(function(err)
+      loading = false
+      core.redraw = true
+      core.error(err or "Error installing addons.")
+    end)
+  end,
+  ["welcome:open-plugin-manager"] = function()
+    command.perform("plugin-manager:show")
+    terminate_welcome()
+  end,
+  ["welcome:dismiss"] = function()
+    terminate_welcome()
+  end
+})
+
+return {  }
```

Comment:
Should probably just not return anything.

Reply:
OK.